addons/mail/controllers/main.py

```diff
@@ -136,38 +136,64 @@ def _redirect_to_record(cls, model, res_id, access_token=None, **kwargs):
         url = '/web?#%s' % url_encode(url_params)
         return werkzeug.utils.redirect(url)
 
+    @http.route('/mail/thread/check_write_access', methods=['POST'], type='json', auth='user')
```

Comment:
Write access allows to post, but that's not the only way of posting. You have notably

  * _mail_post_access allowing to post notably when having read access;
  * the right to reply to messages you have been notified of;

Second one is complicated to simulate I think as using the chatter you don't really answer to a given message. But the first one should be checked according to rules.

You may see ``check_access_rule`` of MailMessage model for more details :)